api/v1beta1/common_types.go

```diff
@@ -164,6 +164,14 @@ type ServiceOptions struct {
 	// Labels to be added for the Service.
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+
+	// Supports "ClientIP" and "None". Used to maintain session affinity. Enable client IP based session affinity. Must be ClientIP or None. Defaults to None
+	// +optional
```

Comment:
```suggestion
	// +optional
	// +kubebuilder:default=None
	// +kubebuilder:validation:Enum=None,ClientIP
```

So we can actually put the default, and the enum constraints into the CRD itself using [Kubebuilder CRD markers](https://book.kubebuilder.io/reference/markers/crd-validation.html)!

Reply:
Ah, neat code-gen